main.py

```diff
@@ -224,10 +216,139 @@ def callback(ch, method, properties, body, ctx):
+    So we query all the objects with the media ID of the fragments.
+    Of those we filter out the real fragments. We should end up with only the
+    collaterals. These will be deleted one by one. Finally, delete the essence.
+    """
+
+    mediahaven_service = MediahavenService(ctx)
+
+    # Query MH with the s3_bucket en s3_object_key
+    s3_bucket = get_from_event(event, "bucket")
+    s3_object_key = get_from_event(event, "object_key")
+    query_params = [
+        ("s3_object_key", f'"{s3_object_key}"'),
+        ("s3_bucket", f'"{s3_bucket}"'),
+    ]
+    try:
+        result = mediahaven_service.get_fragment(query_params, or_params=False)
+    except (HTTPError, RequestException) as error:
+        log.error(
+            f"Error when querying MH fragment with s3 bucket: {s3_bucket} and object key: {s3_object_key}",
+            error=error,
+        )
+        return False
+
+    if not result["MediaDataList"]:
+        log.info(f"No media object found with s3 bucket: {s3_bucket} and object key: {s3_object_key}")
+        return False
+
+    log.info(f"Removing media object with s3 bucket: {s3_bucket} and object key: {s3_object_key}")
+    items = result["MediaDataList"]
+
+    # Collect the Media ID (and Fragment ID) of the fragments. These will be used to remove the collaterals.
+    fragments = {}
+    for item in items:
+        if item["Internal"]["IsFragment"]:
+            fragments[item["Dynamic"]["dc_identifier_localid"]] = item["Internal"]["FragmentId"]
+
+    # Get the Fragment ID of the essence to delete
+    fragment_id_essence = next(
+        item["Internal"]["FragmentId"]
+        for item in items
+        if not item["Internal"]["IsFragment"]
+    )
+
+    if fragments:
+        # Query all the objects with the media IDs of the fragments
+        query_params_media_ids = [
+            ("dc_identifier_localid", f"{media_id}") for media_id in fragments.keys()
+        ]
+        response = mediahaven_service.get_fragment(query_params_media_ids)
+        # Collect the Fragment IDs of the collaterals. The Media ID is used in the delete reason.
+        fragments_collateral = [
+            (item["Internal"]["FragmentId"], item["Dynamic"]["dc_identifier_localid"])
+            for item in response["MediaDataList"]
+            if not item["Internal"]["IsFragment"]
+        ]
+
+        # Delete the collaterals
+        for fragment_collateral in fragments_collateral:
+            # Get the Fragment ID of the fragment to which this collateral is linked to
+            media_id = fragment_collateral[1]
+            linked_fragment_id = fragments.get(media_id)
+            result = delete_media_object(
+                mediahaven_service,
+                fragment_collateral[0],
+                f'This was a collateral with Media ID: "{media_id}" linked to fragment with Fragment ID: "{linked_fragment_id}". Underlying essence has been deleted via s3 delete-object'
+            )
```

Comment:
For VRT, this is a "Media ID". Generally speaking, however, this is a `local_id`.

```suggestion
            local_id = fragment_collateral[1]
            linked_fragment_id = fragments.get(local_id)
            result = delete_media_object(
                mediahaven_service,
                fragment_collateral[0],
                f'Deleted collateral with local_id: "{local_id}" linked to fragment with fragment_id: "{linked_fragment_id}". Essence was deleted via s3 delete-object.'
            )
```